File: solutions/ruby/collatz-conjecture/1/collatz_conjecture.rb
class CollatzConjecture
  def self.steps(number)
    raise ArgumentError if number <= 0
    count = 0
    while number > 1
      number = number.even? ? number / 2 : 3 * number + 1
      count += 1
    end
    count
  end
end

File: solutions/zig/collatz-conjecture/1/collatz_conjecture.zig
pub const ComputationError = error{IllegalArgument};

pub fn steps(number: usize) anyerror!usize {
    if (number <= 0) {
        return ComputationError.IllegalArgument;
    }
    var step = number;
    var count: usize = 0;
    while (step > 1) {
        if (step % 2 == 0) {
            step /= 2;
        } else {
            step = step * 3 + 1;
        }
        count += 1;
    }
    return count;
}
